src/ElasticsearchFactory.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the RollerworksSearch package.
+ *
+ * (c) Sebastiaan Stok <s.stok@rollerscapes.net>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Rollerworks\Component\Search\Elasticsearch;
+
+use Psr\SimpleCache\CacheInterface as Cache;
+use Rollerworks\Component\Search\SearchCondition;
+
+/**
+ * Class ElasticsearchFactory.
+ */
+class ElasticsearchFactory
+{
+    /**
+     * @var Cache
+     */
+    private $cacheDriver;
+
+    /**
+     * @param Cache $cacheDriver
+     */
+    public function __construct(Cache $cacheDriver = null)
+    {
+        $this->cacheDriver = $cacheDriver;
+    }
+
+    /**
+     * Creates a new SqlConditionGenerator for the SearchCondition.
```

Comment:
You may want to check the phpdoc comments :)

Reply:
Oops, will update it now. :+1: